cmd/secret.go

```diff
@@ -0,0 +1,240 @@
+Secret value can be either provided in second argument or through file flag. 
+Use base64 flag if the value has been encoded.
+		`,
+	}
+	secretCmd.Flags().StringVarP(&projectName, "project", "p", conf.GetProject().Name, "project name of optimus managed repository")
+	secretCmd.Flags().StringVarP(&namespaceName, "namespace", "n", conf.GetNamespace().Name, "namespace of deployee")
+	secretCmd.Flags().BoolVarP(&encoded, "base64", "", encoded, "create secret with value that has been encoded")
+	secretCmd.Flags().StringVarP(&filePath, "file", "f", filePath, "provide file path to create secret from file instead")
+
+	secretCmd.RunE = func(cmd *cli.Command, args []string) error {
+		secretName, err := getSecretName(args)
+		if err != nil {
+			return err
+		}
+
+		secretValue, err := getSecretValue(args, filePath, encoded)
+		if err != nil {
+			return err
+		}
+
+		registerSecretReq := &pb.RegisterSecretRequest{
+			ProjectName:   projectName,
+			SecretName:    secretName,
+			Value:         secretValue,
+			NamespaceName: namespaceName,
+		}
+		return registerSecret(l, conf, registerSecretReq)
+	}
+	return secretCmd
+}
+func secretUpdateSubCommand(l log.Logger, conf config.Provider) *cli.Command {
+	encoded := false
+
+	var (
+		projectName   string
+		namespaceName string
+		filePath      string
+	)
+
+	secretCmd := &cli.Command{
+		Use:     "update",
+		Short:   "update secret",
+		Example: "optimus secret update sample_secret secret_value",
+		Long: `
+This operation takes secret name as its first argument.
+Secret value can be either provided in second argument or through file flag.
+Use base64 flag if the value has been encoded.
+		`,
+	}
+	secretCmd.Flags().StringVarP(&projectName, "project", "p", conf.GetProject().Name, "project name of optimus managed repository")
+	secretCmd.Flags().StringVarP(&namespaceName, "namespace", "n", conf.GetNamespace().Name, "namespace of deployee")
+	secretCmd.Flags().BoolVarP(&encoded, "base64", "", encoded, "create secret with value that has been encoded")
+	secretCmd.Flags().StringVarP(&filePath, "file", "f", filePath, "provide file path to create secret from file instead")
+
+	secretCmd.RunE = func(cmd *cli.Command, args []string) error {
+		secretName, err := getSecretName(args)
+		if err != nil {
+			return err
+		}
+
+		secretValue, err := getSecretValue(args, filePath, encoded)
+		if err != nil {
+			return err
+		}
+
+		updateSecretReq := &pb.UpdateSecretRequest{
+			ProjectName:   projectName,
+			SecretName:    secretName,
+			Value:         secretValue,
+			NamespaceName: namespaceName,
+		}
+		return updateSecret(l, conf, updateSecretReq)
+	}
+	return secretCmd
+}
+
+func getSecretName(args []string) (string, error) {
+	if len(args) < 1 {
+		return "", errors.New("secret name is required")
+	}
+	if strings.HasPrefix(args[0], models.SecretTypeSystemDefinedPrefix) {
+		return "", errors.New(fmt.Sprintf("secret name cannot be started with %s", models.SecretTypeSystemDefinedPrefix))
+	}
```

Comment:
we should have this check server side

Reply:
@kushsharma we are not doing it on the server side because people should be able to create user-defined and system-defined secrets through API, while through CLI it is only user-defined secrets.

Sure we can still validate it on the server-side, but we may need to add a field to differentiate those 2 secret types. what do you think?

Reply:
Make sense, let's keep it in CLI only.